docs/development/localnet-with-kubernetes.md

```diff
@@ -0,0 +1,69 @@
+# LocalNet on Kubernetes
+
+We are developing our own Kubernetes operator to manage v1 workloads both internally and for the community to make it easier to deploy pocket v1. While the operator is still in development, we can already utilize it to run our local networks locally on Kubernetes. This guide will show you how to do that.
+
+- [LocalNet on Kubernetes](#localnet-on-kubernetes)
+    - [Dependencies](#dependencies)
+    - [Running the localnet](#running-the-localnet)
+    - [Stopping / cleaning up the resources](#stopping--cleaning-up-the-resources)
+    - [Interaction with the localnet](#interaction-with-the-localnet)
+    - [How does it work?](#how-does-it-work)
+    - [Troubleshooting](#troubleshooting)
+    - [How to change configuration files](#how-to-change-configuration-files)
+
+### Dependencies
+
+* [tilt](https://docs.tilt.dev/install.html)
+* Kubernetes cluster ([different options available](https://docs.tilt.dev/choosing_clusters.html)), please make sure to run 1.23 version or older, until [this issue is resolved](https://github.com/zalando/postgres-operator/issues/2098) - otherwise Postgres DB is not going to be provisioned.
+  * `kubectl` is also required to be installed and configured to access the cluster, but that should happen automatically when you install kubernetes cluster locally.
+  * `helm` is required to template the yaml manifests for the observability stack. Here are the instructions on how to install it: https://helm.sh/docs/intro/install/.
+* pocket kubernetes operator codebase in `../pocket-operator` directory, relative to the pocket v1 codebase. If you don't have it, tilt will try to pull it from the `main` branch on first launch.
```

Comment:
Why not make `pocket-operator a git submodule`?

Then we won't have a weird setup and it'll be easier to automate.

Reply:
From my experience, submodules are a huge PITA when used on repos where multiple people are collaborating. Though I agree, that would give us an ability to tie configuration files versions and logic from pocket-operator to the v1 repo. I just fear that managing a submodule might become a nightmare.

I mentioned that above - I won't be surprised if we remove pocket-operator from LocalNet and will use Kubernetes directly (with Tilt controlling the whole lifecycle). So I guess it doesn't matter if we use submodules or not - they are not going to be a deciding factor if operator sticks for LocalNet or not. :) 

Do you see any issues with using submodule? I can swap Tiltfile code with submodule.